### DISCUSSION TITLE
What happened to Process Hacker?

### DISCUSSION THREAD

**Discussion by ags1234** [winsiderss/systeminformer#2275] [General Discussion]
Why was the name of the project changed?
Process Hacker was a good name and well known already.
ProcessHacker was of a higher quality and efficiency software.

ProcessHacker-x64-2.39 unpacked is 3,884,184 bytes.
SystemInformer-x64-3.1.24298 unpacked is 17,254,949 bytes.

A 4,4 increase in size that is not backed up by the same progression in features.

**Comment by Vasilich** [comment]
ags1234,
Any proof of "ProcessHacker was of a higher quality"?
Have you ever compared the content of folders insoide SystemInformer and ProcessHacker?
Haven't  you seen all the new features in SytemInformer?
Have you read about false positives for many antivirus software now ant then for ProcessHacker? Have you read about driver signing and digital signatures for every executable in SYstemInformer to prevent false positives?

Learn to inform yourself before make false judgements and offensive comments.

**Reply by ags1234** [reply]
First of all, I posted this in Discussions section. It is a simple discussion, with no intention of being inflamatory.

It looks like ProcessHacker changed owners and that caused substantial changes. 

Regarding the false judgements
If you read my first post, you will see that the numbers talk for themselves. I did not invent those numbers. Anyone can reproduce the results.
It is reasonable to think that software having fewer files and fewer megs, while still doing what was supposed to, is more efficient. Therefore of a higher overall quality.
SystemInformer has improvements, but those improvements also contributed to a 4,4 increase in size. I hope SystemInformer does not exponentially increase in size as development continues. If it will become big, it will become slow. Having a slow task manager is not desirable.

Going from VS 2015 (ProcessHacker) to VS 2022 (SystemInformer) to compile the code probably did contribute a lot to the increase in size. ProcessHacker was smaller and faster than other task managers. It would be nice if SystemInformer would inherit these values, while continuing to improve.

Regarding the "offensive comments", I noticed you are using the plural, but I don't know what are those multiple comments you are referring to. In case you are referring to my answer to what @loef4190 posted above, well, his comment was deleted and cannot be seen anymore. I'll repost it here: "There are a lot of untrue recommendations and impostors, its hard to tell who is legit. If you have been scammed by fake crypto investors, lost money to scammers, I refer you to contact (hack4net8@gmail .com) they will surely help you out. Took me long to find them". Out of context, out of place, not answering the questions asked. Looked a lot like an attempted scam, too. As I said, a shit comment. Singular.

Maybe someone finds this useful. It is just an insight from someone who cares enought to spend time giving it.

**Reply by Vasilich** [reply]
Well, let's compare installed versions of PH and SI:
----
PH v3.0.9480.4953 - 9 800 001 bytes,
  * non-binary files (readme, license, counters, manifest, etc.): 366 185 bytes
  * all binary signature files:  64 bytes
  * driver file (kprocesshacker.sys):45 208 bytes
  * ProcessHacker.exe: 2 761 728 bytes
  * peview.exe: 1 142 272 bytes
  * all x64 plugins (except ExtendedTools): 2 363 904 bytes
      ExtendedTools.dll: 395 776 bytes
----
SI v3.1.24318.0 - 17 293 734 bytes.
  * non-binary files (readme, license, counters, icon, etc.): 375 386 bytes
  * all binary signature files:  8 704 bytes
  * debug API package (explanation for its necessity is here https://github.com/winsiderss/systeminformer/discussions/2312 ): 3 061 109 bytes
  * driver files, spltited for signature and adaptation to continuous Windows kernel changes purposes (SystemInformer.sys, ksi.dll, ksidyn.bin): 378 819 bytes
  * SystemInformer.exe: 3 379 416 bytes
  * peview.exe: 1 688 424 bytes
  * all x64 plugins (except ExtendedTools): 2 853 744 bytes
      ExtendedTools.dll (most notable features increase): 1 350 600 bytes
----
First - there is **no** increase in **4** times, as you wrote before.
As you see, there are some changes, that are based on necessary structural changes (driver, signatures) and adaptation to changing world (driver split), that make SystemInformer possible to resist to changes from third-party software for malisious purposes (and reduce count of false positives for antivirus software - you surely remember that PH had much more problems with those). Also there are a lot of changes to increase functionality of SI - e.g. tabs for devices and firewall, extended information for many areas (like S.M.A.R.T info incl NVMe devices, etc.).

Do you still think that it doesn't explain the increase in size of software without reducing its quality?
Or do you want to compare functionality of PH and SI first, and after that change your mind about judjement of software quality based on its binary size only, counting all the enhancements?

**Reply by ags1234** [reply]
Thanks for the detailed answer.

> Well, let's compare installed versions of PH and SI

Why change the versions? In my first post I was referring to unpacked versions from zip archives, not the installed ones.

> there is no increase in 4 times, as you wrote before.

Well, I redid the math and there is about a 4.4 increase in size. See the sizes below.

![2024-12-27_155614](https://github.com/user-attachments/assets/69d30fce-1fd8-4c98-b256-12bbf17869dd)

It's just that there is a trend today among the developers to use whatever technologies and frameworks to build software. In most cases one could suspect it's the use of framework over skill. This makes the software bloated and slow. So users end up with software that have a few dialogs and a few buttons and no particularly large resources, but goes over 100 MiB.

A more concrete example to illustrate what I am trying to say is MediaMonkey.
MediaMonkey version 4 is about 56 MiB.
MediaMonkey version 5 is about 354 MiB (!!).
MediaMonkey 4's size could be considered normal for the amount of features it has.
In version 5 they made it into a browser, based it on Chromium and the size exploded. Version 4 was already a very good player with all the features most people need and I cannot possibly think that version 5, that has the same features is in any way better at 6.6 increase in size.

On the other hand, look at Total Commander, Double Commander or WinRar, to name just a few that come to mind now. Small disk size, small RAM consumption, extremely optimized and fast, packed full of options. They provide a lot of value per mebibyte. So yeah, that results in a higher quality software.

Anyway, please know that SI is still very good and all the work put into it is appreciated.
I was just trying to raise some awareness on using the technology responsible and as efficiently as we can.

**Reply by Vasilich** [reply]
you have compared the PH 2.39 only folder with x64 version with SI 3.xx bundle x32/x64 version. For comparison to be fair you need to compare either full folder of PH, or delete x32 folder from SI before comparison (and exclude all text files as you have done for PH).

![image](https://github.com/user-attachments/assets/9b4159d4-4530-4901-a6e6-9fdde94697de)

**Reply by ags1234** [reply]
Is that Total Commander I see? ;)

**Reply by ags1234** [reply]
> For comparison to be fair you need to compare either full folder of PH, or delete x32 folder from SI before comparison (and exclude all text files as you have done for PH).

Well I am not sure I fully understand what you think I am doing wrong. I compare exactly what developers released into the wild.

I downloaded PH and SI and I compared the x64 versions of both, extracted from the following archives:
processhacker-2.39-bin.zip
systeminformer-3.1.24298-release-bin.zip

From `processhacker-2.39-bin.zip` I take the directory `x64`, exactly as the developers left it.
From `systeminformer-3.1.24298-release-bin.zip` I take the directory `amd64`, exactly as the developers left it.
The SI archive also contains the directories `arm64` and `i386`, but those are not the object of this comparison, because I compared x64 versions.

I have no idea why `amd64` from SI also contains the 32 bit version again. But if it does, it is only fair to be included in comparison. I someone from the developers cares about the size at all, they should remove it, not me.

Also, I have not excluded any text files, I compared full directories as found in the archives.

PS:
However, since you suggested it...  :))
If I remove `x86` from `amd64` from SI, goes down to 13.056.981 bytes from 17.286.075 bytes.
And PH is 3.884.184 bytes.

**Reply by Vasilich** [reply]
answer to "why amd64 from SI also contains the 32 bit version": https://github.com/winsiderss/systeminformer/discussions/2347

**Comment by dmex** [comment]
> Well, I redid the math and there is about a 4.4 increase in size. See the sizes below.

One of those zip files is using Deflate compression and the other is using LZMA2 compression. The PE file alignment of every binary image was also changed from 512 to match the native page alignment 4096.

> On the other hand, look at Total Commander, Double Commander or WinRar, to name just a few that come to mind now. Small disk size, small RAM consumption, extremely optimized and fast, packed full of options. They provide a lot of value per mebibyte. So yeah, that results in a higher quality software.

Those applications have the luxury of a different threat landscape than we do. We're targeted by a large amount of malicious software that wants to conceal system activity and malware/spyware/etc.. We also spend a considerable amount of time reverse engineering Windows because Microsoft enforces severe restrictions on the development of third party task managers compared to Task Manager and Process Explorer. 

> raise some awareness on using the technology responsible and as efficiently as we can.

We don't have the luxury of using frameworks because task managers are expected/required to maintain system performance and stability, especially in critical situations where the system has no resources available, under stress or malfunctioning and terminate or troubleshoot issues with the system. 

Microsoft still maintains the original taskmgr even on the latest versions of Windows for the same reason: 
![image](https://github.com/user-attachments/assets/2235317a-6ee8-4468-bbd1-e7ae58418da1)

**Comment by Vasilich** [comment]
Counting all arguments (including links to other 2 discussions and direct answer from @dmex ) I believe the discussion can be closed because all the differences leading to binary size increase were cleared (at least for me).
@ags1234 isn't it?

**Comment by ags1234** [comment]
@dmex 
> One of those zip files is using Deflate compression and the other is using LZMA2 compression.

The names of the arhives were provided just to help in reproducing my observations. I think compression method of zip files has nothing to do with my observations, since I for the comparison I extracted the files from archive and at no point I compare archives. 

@Vasilich, @dmex
Thanks you for taking the time to reply and provide your insight on the matter.
